docs/DeveloperGuide.adoc

```diff
@@ -227,8 +239,76 @@ image::ResumeEditCommandAlt1.png[]
+
+==== Current Implementation
+The `tagpull` command is facilitated by `TagPullCommand`, which extends `Command`. Therefore, like any other `Command` classes, it will have an `execute` method.
+
+Given below is an example usage scenario and how the `tagpull` works at each step.
+
+Step 1. The user launches the application, and uses the `add` command to add several `Resume`, `Internship`, `Project`, and `Skill` items.
+
+Step 2. The user executes `tagpull 1 #/ tech` command to add all items that have been tagged with `tech` to the first `Resume` in the list of `Resume` items.
+
+Step 3. This calls `ResumeBookParser#parseCommand()`, which would create a new `TagPullCommandParser` object and call the `TagPullCommandParser#parse()` method.
+
+Step 4. A new `TagPullCommand` object is created. It contains the index of the `Resume` that is to be modified, and the tags of the items which the user wants to add. In this example, it will only have the `tech` tag.
+
+Step 5. The `TagPullCommand#execute()` method is called with the current `model`. A copy of the `Resume` is created and all the items with matching tags are retrieved from `model`. The content of the copied `Resume` is updated to now contain all the items with matching tags, on top of existing ones.
+
+Step 6. A new `TagPullCommand` object, which contains the edited copy of the `Resume`, is created and returned.
+
+The following sequence diagram shows how `tagpull` works:
+
+// TODO: Add Sequence Diagram
+
+==== Design Considerations
+===== Aspect: Integrating Tag Pull to Resume Edit
+* **Alternative 1 (current choice):** Separating `redit` and `tagpull`
+
+** Pros: Separation of concerns. `redit` handles updates of `Resume` using indices and `tagpull` handles updates of `Resume` using tags.
+** Cons: There is some code duplication due to similarities in behaviour.
+
+* **Alternative 2:** Combine `redit` with `tagpull`, making use of `#/` as prefix for `redit` command
+
+** Pros: It may be intuitive for user to only have a single command that handles modification of `Resume`
+** Cons: The implementation of `redit` becomes much more complicated as various combinations of input has to be considered. For example, considerations of what the expected behaviour should be if indices and tags are given.
+
+****
+*Conclusion:* We decided to separate the two commands in order to have a simpler implementation of the commands. By separating the two, there is a separation of concerns and there is no need to consider the behaviour when both indices and tags are given as arguments.
+
+It may also save the user some confusion since the `redit` feature specifically handles only updates using indices while the `tagpull` feature handles only updates using tags.
+
+// We also considered a use case: "As a user, I want to easily have items of the same category (tags) in my resume". We felt that users will find it easier to use `tagpull` to retrieve items of the same tags ok idk what im smoking hahahha help pls
```

Comment:
A plea for help

Reply:
Cons: it defeated the semantic implication of `tagpull` and what we have envisioned for its purpose. We intend `tagpull` to be a potent command at a small cost, and it would be more useful standalone than included in some other command.

Cons: `tagpull`  is a cool name and easy to remember.

Reply:
Thanks for the suggestions! After further discussions, I think I may have had enough points to raise up in this section.